### PR TITLE
GameObject: Implement on Engine

### PR DESCRIPTION
**Pull request by AstraLuma**
This does two things:

* Moves the management of children out of `Engine`
* Adds `GameObject` to `Engine`.

Part of #492

### CODE REVIEW

**Comment by AstraLuma** [comment]
~Currenly viztests are failing for me in Wacky Ways, so I haven't tested on those.~ yup it works.

File: ppb/engine.py [hunk 1]
```diff
@@ -39,7 +46,170 @@ def _get_handler_name(txt):
...
+
+        Note that Scenes and Systems have special restrictions.
+
+        Examples: ::
+
+            children.add(MyObject())
+
+            children.add(MyObject(), tags=("red", "blue")
+        """
+        # Ugh, this is a copy of the implementation in Children.
```
Comment (pathunstrom) [inline ppb/engine.py:116]: Not for this PR but we could consider a "validate_child" helper function in `Children` that can be called in implementations of `add` and overridden for things like the check on line 122.

Need to consider the implementation considering we don't want to rewrite 131-134 either.

File: ppb/engine.py [hunk 2]
```diff
@@ -70,25 +240,23 @@ def __init__(self, first_scene: Type, *,
            consequences. Consider passing via systems parameter instead.
         """
 
-        super(GameEngine, self).__init__()
+        super().__init__()  # FIXME: This is breaking the GameObject protocol
```
Comment (pathunstrom) [inline ppb/engine.py:244]: O.o How is this breaking the protocol?

Comment (AstraLuma) [reply ppb/engine.py:244]: Part of `GameObject` is that `__init__()` takes initial properties as kwargs.

```python
ge = GameEngine(spam='eggs')
ge.spam  # AttributeError
```

Comment (pathunstrom) [reply ppb/engine.py:244]: Aha. I should play with run because we might be able to make that happen.

Comment (AstraLuma) [reply ppb/engine.py:244]: It shouldn't be at the `run()` level? like, that's a convenience to make it easier to stand up an Engine.

Comment (pathunstrom) [reply ppb/engine.py:244]: I mean, we convert the engine to only kwargs, run keeps its current API, we use run as the translation layer between.

**Review by pathunstrom — APPROVED** [review]
bors r+

**Pull request merged**